sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -502,6 +498,80 @@ export class CryptographyClient {
     }
   }
 
+  private static encodeLength(length: number): Uint8Array {
+    if (length <= 127) {
+      return Uint8Array.of(length);
+    } else if (length < 256) {
+      return Uint8Array.of(0x81, length);
+    } else if (length < 65536) {
+      return Uint8Array.of(0x82, length >> 8, length & 0xff);
+    } else {
+      throw new Error("Unsupported length to encode");
+    }
+  }
+
+  private static encodeBuffer(buffer: Uint8Array, bufferId: number): Uint8Array {
+    if (buffer.length == 0) {
+        return buffer;
+    }
+
+    let result = new Uint8Array(buffer);
+    
+    // If the high bit is set, prepend a 0
+    if ((result[0] & 0x80) === 0x80) {
+        let array = new Uint8Array(result.length + 1);
+        array[0] = 0;
+        array.set(result, 1);
+        result = array;
+    }
+
+    // Prepend the DER header for this buffer
+    let encodedLength = CryptographyClient.encodeLength(result.length);
+
+    let totalLength = 1 + encodedLength.length + result.length;
+
+    let outputBuffer = new Uint8Array(totalLength);
+    outputBuffer[0] = bufferId; 
+    outputBuffer.set(encodedLength, 1);
+    outputBuffer.set(result, 1 + encodedLength.length);
+
+    return outputBuffer;
+  }
+
+  private static convertJWKtoPEM(key: JsonWebKey): string {
+    if (!key.n || !key.e) {
+      throw new Error("Unsupported key format for local operations");
+    }
+    let encoded_n = CryptographyClient.encodeBuffer(key.n, 0x2); // INTEGER
+    let encoded_e = CryptographyClient.encodeBuffer(key.e, 0x2); // INTEGER
+
+    let encoded_ne = new Uint8Array(encoded_n.length + encoded_e.length);
+    encoded_ne.set(encoded_n, 0);
+    encoded_ne.set(encoded_e, encoded_n.length);
+
+    let full_encoded = CryptographyClient.encodeBuffer(encoded_ne, 0x30); //SEQUENCE
+
+    let buffer = Buffer.from(full_encoded).toString("base64");
+
+    let beginBanner = "-----BEGIN RSA PUBLIC KEY-----\n";
+    let endBanner = "-----END RSA PUBLIC KEY-----";
+
+    let outputString = beginBanner;
+    let lines = buffer.match(/.{1,64}/g);
+    
+    if (lines) {
+      for (let line of lines) {
+        outputString += line;
+        outputString += "\n";
+      }
```

Comment:
Ohh, interesting. Is this a necessary part of the format?

Reply:
This comes from the PEM standard.  If you're curious, here's the quote from the RFC:

```
To represent the encapsulated text of a PEM message, the encoding
   function's output is delimited into text lines (using local
   conventions), with each line except the last containing exactly 64
   printable characters and the final line containing 64 or fewer
   printable characters.
```

I'll put it in the file to help other people understand why it's written that way